### PR TITLE
IDENTITY-6654:Improve unit tests coverage for OIDC Session management

### PR DESCRIPTION
**Pull request by piraveena**
Fixed IDENTITY-6676 also

### CODE REVIEW

File: components/org.wso2.carbon.identity.oidc.session/pom.xml [hunk 1]
```diff
@@ -78,6 +78,32 @@
             <groupId>org.wso2.orbit.com.nimbusds</groupId>
             <artifactId>nimbus-jose-jwt</artifactId>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/pom.xml]: This should have the 'test' scope.

File: components/org.wso2.carbon.identity.oidc.session/pom.xml [hunk 2]
```diff
@@ -78,6 +78,32 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/pom.xml]: Is there a particular reason to use two test frameworks within the same repo?

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+public class OIDCSessionManagementUtilTest {
+
+    private static final String CLIENT_ID = "u5FIfG5xzLvBGiamoAYzzcqpBqga";
+    private static final String CALLBACK_URL = "http://localhost:8080/playground2/oauth2client";
+    private static final String OPBROWSER_STATE = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
+    private static final String SESSION_STATE = "18b2343e6edaec1c8b1208169ffa141d158156518135350be60dfbf6f41d340f" +
+            ".W2Gf-RAzLUFy2xq_8tuM6A";
+    String responseType[] = new String[]{"id_token", "token", "code" };
+
+
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: unnecessary space

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
+package org.wso2.carbon.identity.oidc.session.util;
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: missing license. Please configure you IDE according to the wso2 template. :)

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+    @Mock
+    HttpServletRequest httpServletRequest;
+
+    @Test
+    public void testGetSessionStateParam() throws Exception {
+
+        String state = OIDCSessionManagementUtil.getSessionStateParam(CLIENT_ID, CALLBACK_URL, OPBROWSER_STATE);
+        Assert.assertNotNull(state, "This is empty");
+
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: unnecessary space.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                {url3, SESSION_STATE, responseType[2], actual3},
+                {url2, SESSION_STATE, responseType[2], actual4},
+                {url2, "", responseType[2], url2}
+        };
+
+    }
+
+    @Test(dataProvider = "provideDataFortestAddSessionStateToURL")
+    public void testAddSessionStateToURL(String url, String sessionState, String responseType, String actual) throws
+            Exception {
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: Any particular reason to throw a generic exception?

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mock);
+
+        String urlReturned = OIDCSessionManagementUtil.addSessionStateToURL(url, sessionState, responseType);
+        Assert.assertEquals(urlReturned, actual, "Invalid returned value");
+
+    }
+
+    @Test
+    public void testAddSessionStateToURL1()
+            throws Exception {
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: Same as above

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({OAuthServerConfiguration.class})
+public class OIDCSessionManagementUtilTest {
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: missing class comments.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    @Test
+    public void testGetSessionStateParam() throws Exception {
+
+        String state = OIDCSessionManagementUtil.getSessionStateParam(CLIENT_ID, CALLBACK_URL, OPBROWSER_STATE);
+        Assert.assertNotNull(state, "This is empty");
+
+    }
+
+    @DataProvider(name = "provideDataFortestAddSessionStateToURL")
+    public Object[][] provideDataFortestAddSessionStateToURL() {
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: missing method comments.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mock);
+
+        String state = OIDCSessionManagementUtil.addSessionStateToURL(url, CLIENT_ID, CALLBACK_URL, obpscookie,
+                responseType[1]);
+        Assert.assertNotNull(state, "This is empty");
+
+    }
+
+    @DataProvider(name = "provideDataForTestGetOPBrowserStateCookie")
+    public Object[][] provideDataForTestGetOPBrowserStateCookie() {
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: missing method comments.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        };
+    }
+
+    @Test(dataProvider = "provideDataForTestGetOPBrowserStateCookie")
+    public void testGetOPBrowserStateCookie(Object cookie, Object expectedResult) throws Exception {
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getCookies()).thenReturn(new Cookie[]{(Cookie) cookie});
+        Assert.assertEquals(OIDCSessionManagementUtil.getOPBrowserStateCookie(request), expectedResult);
+
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: additional space.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+}
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: Please add a new line at the end.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getCookies()).thenReturn(new Cookie[]{(Cookie) cookie});
+        Assert.assertEquals(OIDCSessionManagementUtil.getOPBrowserStateCookie(request), expectedResult);
+
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: Why don't we import the package org.powermock.modules.testng?

File: components/org.wso2.carbon.identity.oidc.session/src/test/resources/testng.xml [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Identity-Oidc-Session">
+    <test name="Util-Tests" preserve-order="true" parallel="false">
+        <classes>
+            <class name="org.wso2.carbon.identity.oidc.session.util.OIDCSessionManagementUtilTest" />
+        </classes>
+    </test>
+</suite>
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.oidc.session/src/test/resources/testng.xml]: new line at the end is needed.

**Comment by piraveena** [comment]
Fixed with requested changes

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtil.java [hunk 15]
```diff
@@ -147,14 +147,15 @@ public static String addSessionStateToURL(String url, String clientId, String rp
...
+     * @return CookieString url, String clientId, String rpCallBackUrl,
+                                              Cookie opBrowserStateCookie, String responseType
      */
     public static Cookie getOPBrowserStateCookie(HttpServletRequest request) {
 
         Cookie[] cookies = request.getCookies();
         if (cookies != null) {
             for (Cookie cookie : cookies) {
-                if (cookie.getName().equals(OIDCSessionConstants.OPBS_COOKIE_ID)) {
+                if (cookie != null && cookie.getName().equals(OIDCSessionConstants.OPBS_COOKIE_ID)) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtil.java]: In the next PR better to do you code fixes like this in separate commits.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({OAuthServerConfiguration.class, OIDCSessionManagementConfiguration.class,
+        IdentityCoreServiceComponent.class, IdentityUtil.class, OAuthServerConfiguration.class})
+/***
+ * This class does unit test coverage for OIDCSessionManagementUtil class
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/util/OIDCSessionManagementUtilTest.java]: Unit tests for OIDCSessionManagementUtil should be enough right?
You are mentioning class twice :smile:

**Comment by piraveena** [comment]
Fixed with requested changes

**Pull request merged**